test/uk/gov/hmrc/tradergoodsprofilesrouter/controllers/RemoveRecordControllerSpec.scala

```diff
@@ -82,21 +83,27 @@ class RemoveRecordControllerSpec extends PlaySpec with MockitoSugar with BeforeA
       status(result) mustBe NO_CONTENT
     }
 
-    "not validate headers when drop2 feature flag is enabled" in {
+    "not validate Client Id header when isClientIdHeaderDisabled flag is enabled" in {
+      when(appConfig.isClientIdHeaderDisabled).thenReturn(true)
+
       when(mockService.removeRecord(any, any, any)(any))
         .thenReturn(Future.successful(Right(NO_CONTENT)))
-      when(appConfig.isDrop2Enabled).thenReturn(true)
 
-      val result = controller.remove(eori, recordId, actorId)(FakeRequest())
+      val result = controller.remove(eori, recordId, actorId)(FakeRequest().withHeaders(validHeaders.filterNot {
```

Comment:
just a suggestion to make the code more clear. Why do not just create a function that return the header you want and named it properly. In this way we remove the filterNot on the Seq and it will be more simple to read

Reply:
Refactored to a simpler filtering easy to read and quickly understand 
```scala
val headersWithoutClientId: Seq[(String, String)] = validHeaders.filterNot {
        case (name, _) => name.equalsIgnoreCase("X-Client-ID")
      }

val result = controller.remove(eori, recordId, actorId)(
        FakeRequest().withHeaders(headersWithoutClientId: _*)
      )
   ```